### DISCUSSION TITLE
Possible to dynamically generate and show widgets?

### DISCUSSION THREAD

**Discussion by cj-fbr** [filamentphp/filament#10412] [Help]
I am working on an idea that would require me to be able to dynamically generate chart widgets and display them on a dashboard, where those charts would be dependent on a user choosing a filter.

For (a bad) example:

Consider a dashboard with a filter form.  On that form is a select field where the user can select one of two options: Fruit or Vegetables.

If they select fruit, I want to be able to hit the database, grab my 'n' fruits and generate a chart widget for each fruit showing some relevant data for each fruit type.  The same thing would apply if the user chose Vegetables.

So maybe today, the Fruit option would generate 3 charts (something for strawberries, plums, and apples), but tomorrow an administrator might create a new fruit entry for grapes so the dashboard would now need to generate a fourth chart.

Is this possible with filament out of the box?  My best guess at an approach right now would be to create a custom widget that would hold the charts, but I'd need to create the charts from scratch (without using the chart type widget in filament).

**Comment by LeonOtto01** [comment]
I'm currently asking myself the same question. Did you find a way to solve it?
